torchvision/models/swin_transformer.py

```diff
@@ -362,7 +436,11 @@ def __init__(
         num_features = embed_dim * 2 ** (len(depths) - 1)
         self.norm = norm_layer(num_features)
         self.avgpool = nn.AdaptiveAvgPool2d(1)
-        self.head = nn.Linear(num_features, num_classes)
+        self.head: Optional[nn.Module] = None
+        if num_classes is not None:
+            self.head = nn.Linear(num_features, num_classes)
+        else:
+            self.head = None
```

Comment:
I'm not sure about this change. It's not an idiom we support at TorchVision, I would revert this one.

Reply:
Ok, agree that this is not part of this PR